src/MultiOwnable.sol

```diff
@@ -1,118 +1,206 @@
+    ///      for the extra 0 bytes.
+    ///
+    ///      In the context of checking whether something was signed by an owner this
+    ///      means that the signature also needs to include the id of the owner.
+    ///
+    ///      The underlying raw bytes can either be:
+    ///         - An abi encoded ethereum address
+    ///         - The abi encoded public key (x, y) coordinates when using passkey.
+    mapping(uint256 index => bytes owner) ownerAtIndex;
+    /// @dev Mapping of raw bytes accounts to booleans indicating whether or not the
+    ///      account is an owner of this contract.
+    mapping(bytes account => bool isOwner_) isOwner;
 }
 
 /// @title Multi Ownable
+///
 /// @notice Auth contract allowing multiple owners, each identified as bytes.
+///
 /// @author Coinbase (https://github.com/coinbase/smart-wallet)
 contract MultiOwnable {
-    /// keccak256(abi.encode(uint256(keccak256("coinbase.storage.MultiOwnable")) - 1)) & ~bytes32(uint256(0xff))
+    /// @dev Slot or the `MultiOwnableStorage` struct in storage.
+    ///      Computed from: keccak256(abi.encode(uint256(keccak256("coinbase.storage.MultiOwnable")) - 1)) & ~bytes32(uint256(0xff))
     bytes32 private constant MultiOwnableStorageLocation =
         0x97e2c6aad4ce5d562ebfaa00db6b9e0fb66ea5d8162ed5b243f51a2e03086f00;
 
+    /// @notice Reverted when the sender is not an owner and is trying to call a privileged function.
     error Unauthorized();
+
+    /// @notice Reverted when trying to add an already registered owner.
+    ///
+    /// @param owner The raw abi encoded owner bytes.
     error AlreadyOwner(bytes owner);
+
+    /// @notice Reverted when trying to remove an owner from an index that is empty.
+    ///
+    /// @param index The targeted index for removal.
     error NoOwnerAtIndex(uint256 index);
+
+    /// @notice Reverted when trying to intialize the contracts owners if a provided owner is neither
+    ///         64 bytes long (for passkey) nor a valid address.
+    ///
+    /// @param owner The invalid raw abi encoded owner bytes.
     error InvalidOwnerBytesLength(bytes owner);
+
+    /// @notice Reverted when trying to intialize the contracts owners if a provided owner 32 bytes long
+    ///         but does not fit in an `address` type (`uint160`).
+    ///
+    /// @param owner The invalid raw abi encoded owner bytes.
     error InvalidEthereumAddressOwner(bytes owner);
 
+    /// @notice Emitted when a new owner is registered.
+    ///
+    /// @param index The owner index.
+    /// @param owner The raw abi encoded owner bytes.
     event AddOwner(uint256 indexed index, bytes owner);
+
+    /// @notice Emitted when an owner is removed.
+    ///
+    /// @param index The owner index.
+    /// @param owner The raw abi encoded owner bytes.
     event RemoveOwner(uint256 indexed index, bytes owner);
 
+    /// @notice Access control modifier ensuring the caller is an owner fo this account (or the account itself).
```

Comment:
For two reasons, I think this should be simplified: 
1. It's virtual so the behavior might be changed at some point when overwritten
2. The method `_checkOwner()`'s natspec should describe the cases where _checkOwner would pass or fail

```suggestion
    /// @notice Access control modifier ensuring the caller is an authorized owner
```